src/child_process_supervisor.rs

```diff
@@ -74,11 +76,37 @@ async fn wait_for_child(
     let child = &mut command.spawn().context("Failed to spawn subprocess")?;
     match waited(duration, flag, child.wait()).await {
         Outcome::Timeout => {
+            #[cfg(windows)]
             kill_child_tree(child);
+            #[cfg(unix)]
+            {
+                use tokio::time::sleep;
+                interrupt(child);
+                tokio::select! {
+                    _ = child.wait() => { },
+                    _ = sleep(Duration::from_secs(10)) => {
+                        kill_child_tree(child);
+                        let _ = child.wait().await;
+                    },
+                };
```

Comment:
Would it make sense to also move this into `interrupt`?

Reply:
Done